.github/workflows/zxcron-promote-build-candidate.yaml

```diff
@@ -0,0 +1,201 @@
+##
+# Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##
+
+name: "ZXCron: Promote Build Candidate"
+on:
+  workflow_dispatch:
+  schedule:
+    # Runs Extended Test Suite every three hours
+    - cron: '0 20 * * *'
+
+permissions:
+  actions: read
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  determine-build-candidate:
+    name: Fetch Latest Build Candidate
+    runs-on: network-node-linux-medium
+    outputs:
+      build_candidate_exists: ${{ steps.find-build-candidates.outputs.build_candidate_exists }}
+      build_candidate_commit: ${{ steps.find-build-candidates.outputs.build_candidate_commit }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@f086349bfa2bd1361f7909c78558e816508cdc10 # v2.8.0
+        with:
+          egress-policy: audit
+
+      # Checkout the latest from dev
+      - name: Checkout Code
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: '0'
+          ref: develop
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+
+      - name: Find Build Candidates
+        id: find-build-candidates
+        run: |
+          TAG_PATTERN="xts-pass-*"
+          if [ $(git tag -l "${TAG_PATTERN}") ]; then
+            echo "build_candidate_exists=true" >> $GITHUB_OUTPUT
+            CANDIDATE_TAG=`git tag --list --sort=-version:refname ${TAG_PATTERN} | head -n 1`
+            CANDIDATE_COMMIT=`git rev-list -n 1 ${CANDIDATE_TAG}`
+            git branch --contains ${CANDIDATE_COMMIT} | grep -q develop
+            echo "build_candidate_commit=${CANDIDATE_COMMIT}" >> $GITHUB_OUTPUT
+            echo "### Build Candidate Found" >>  $GITHUB_STEP_SUMMARY
+            echo "build_candidate_commit=${CANDIDATE_COMMIT}" >> $GITHUB_STEP_SUMMARY
+            echo "build_candidate_tag=${CANDIDATE_TAG}" >> ${GITHUB_STEP_SUMMARY}
+            git tag -d $(git tag -l ${TAG_PATTERN})
+            git push -d origin $(git tag -l ${TAG_PATTERN})
```

Comment:
Can we use more long options here, for consistency and self-documenting?

`git tag --delete` -d
`git tag --list` -l
`git push --delete` -d
`git rev-list --max-count` -n
`head --lines` -n
`grep --quiet|--silent` -q

Reply:
Good call, I prefer to use the long options when writing the scripts just kind of rushed through some of these :)